classes/evasys_inviter.php

```diff
@@ -0,0 +1,313 @@
+                }
+            }
+            return $surveys;
+        }
+    }
+
+    public function close_evasys_surveys($courses) {
+        foreach ($courses as $evasysid => $moodlecourse) {
+            $errorsurveys = array();
+            // Get all open evasys Surveys.
+            $surveys = $this->get_evasys_course_surveyids($evasysid, false);
+            foreach ($surveys as $survey) {
+                if (!$this->set_close_task($survey)) {
+                    array_push($errorsurveys, $evasysid);
+                }
+            }
+            if (count($errorsurveys) > 0) {
+                $this->send_failure_warning($moodlecourse, $evasysid, $errorsurveys);
+            }
+        }
+    }
+
+    public function send_failure_warning ($moodlecourse, $evasysid, $errorsurveys) {
+        global $USER;
+        $course = get_course($moodlecourse);
+        $userto = evasys_synchronizer::get_assigned_user($course);
+        $userfrom = $USER;
+        $subject = "Automatisches Senden der Evaluationsergebnisse für Moodlekurs $moodlecourse fehlgeschlagen!";
+        $message = "Sehr geehte/r Evaluationskoordinator/in \n\r\n\r";
+        $message .= "Leider gab es ein Problem beim automatischen Versand der Ergebnisse.\r\n";
+        $message .= "Bitte versenden sie die Evaluationsergebnisse für folgende Umfragen: \r\n\r\n";
+        $message .= "\tEvasysveranstaltung: $evasysid \r\n";
+        $message .= "\tEvasysumfrageids: \r\n";
+        foreach ($errorsurveys as $errorsurvey) {
+            $message .= "\t\t -$errorsurvey\r\n";
+        }
+        $message .= "\r\n";
+        $message .= "Mit freundlichen Grüßen \r\n";
+        $message .= "Ihr Learnweb-Support.";
+        email_to_user($userfrom, $userto, $subject, $message);
+    }
+
+    public function open_evasys_surveys($courses) {
+        $surveys = array();
+        foreach ($courses as $course) {
+            // Get all open evasys Surveys.
+                $surveys = array_merge($surveys, $this->get_evasys_course_surveyids($course, false));
+        }
+        $surveys = array_unique($surveys);
+
+        foreach ($surveys as $survey) {
+            $result = $this->soapclient->sendInvitationToParticipants($survey);
+        }
+    }
+
+    public function open_moodle_courses($courseids) {
+        $courses = array();
+        foreach ($courseids as $courseid) {
+            if (self::getmode(get_course($courseid)->category)) {
+                if ($this->push_users_in_moodlecourse($courseid)) {
+                    $courses = array_merge($courses, self::get_evasysids($courseid));
+                }
+            }
+        }
+        $this->open_evasys_surveys($courses);
+        global $DB;
+        if (count($courseids) != 0) {
+            list($where, $params) = $DB->get_in_or_equal($courseids);
+            $DB->execute("UPDATE {block_evasys_sync_courseeval} SET state = 1 WHERE course $where", $params);
+        }
+        foreach ($courseids as $courseid) {
```

Comment:
The logic is not quite right. Currently, `open_evasys_surveys` only works on courses that have `getmode == true`, but events are recorded --and status is updated-- for every single course supplied in `$courseids`.